src/worker.py

```
@@ -993,66 +994,88 @@ async def api_dashboard(req, env):
+        "(SELECT COUNT(*) FROM enrollments WHERE activity_id=a.id AND status='active') AS participant_count,"
         "(SELECT COUNT(*) FROM sessions WHERE activity_id=a.id) AS session_count"
         " FROM activities a WHERE a.host_id=? ORDER BY a.created_at DESC"
     ).bind(user["id"]).all()
+    hosted_rows = res.results or []
+    hosted_ids = [r["id"] for r in hosted_rows]
+    hosted_tags = {}
+    if hosted_ids:
+        placeholders = ",".join("?" * len(hosted_ids))
+        tag_res = await env.DB.prepare(
+            f"SELECT at2.activity_id, t.name FROM tags t JOIN activity_tags at2 ON at2.tag_id=t.id"
+            f" WHERE at2.activity_id IN ({placeholders})"
```

Ruff (0.15.7)

[error] Possible SQL injection vector through string-based query construction

(S608)